src/tags.rs

```diff
@@ -1,76 +1,63 @@
 use super::api::ApiClient;
-use clap::{App, Arg, ArgMatches};
+use structopt::StructOpt;
 use fuzzy_filter::FuzzyFilter;
 use twitch_api2::helix::tags::TwitchTag;
 
-pub fn get_tags_app<'a>() -> App<'a, 'a> {
-    App::new("tags")
-        .about("manipulate a streams tags")
-        .arg(
-            Arg::with_name("locale")
-                .short("i")
-                .long("locale")
-                .default_value("en-us"),
-        )
-        .subcommand(
-            App::new("listall")
-                .arg(
-                    Arg::with_name("long")
-                        .short("l")
-                        .help("use a long listing format"),
-                )
-                .arg(
-                    Arg::with_name("needle")
-                        .index(1)
-                        .help("filter tags with a fuzzy search"),
-                ),
-        )
-        .subcommand(
-            App::new("list")
-                .arg(
-                    Arg::with_name("long")
-                        .short("l")
-                        .help("use a long listing format"),
-                )
-                .arg(
-                    Arg::with_name("broadcaster")
-                        .index(1)
-                        .required(true)
-                        .help("the broadcaster you want to fetch tags for"),
-                        //TODO default to the one running this command
-                )
-                .arg(
-                    Arg::with_name("needle")
-                        .index(2)
-                        .help("filter tags with a fuzzy search"),
-                ),
-        )
+#[derive(StructOpt)]
+#[structopt(about = "manipulate a streams tags")]
+pub struct TagsOptions {
+    #[structopt(short = "i", long, default_value = "en-us")]
+    pub locale: String,
+
+    #[structopt(subcommand)]
+    pub subcommand: TagsSubcommand,
+}
+
+#[derive(StructOpt)]
+pub enum TagsSubcommand {
+    ListAll {
+        #[structopt(flatten)]
+        shared: SharedTagsOptions,
+    },
+    List {
+        #[structopt(flatten)]
+        shared: SharedTagsOptions,
+        broadcaster: String,
+    },
 }
 
-pub async fn tags<'a>(client: ApiClient<'a>, matches: &ArgMatches<'a>) {
-    match matches.subcommand() {
-        ("listall", Some(sub_m)) => {
+#[derive(StructOpt)]
+pub struct SharedTagsOptions {
+    #[structopt(short)]
+    long: bool,
+    #[structopt(short, long)]
+    search_string: Option<String>,
```

Comment:
I would prefer having this as positional argument since I'd rarely use list-all without it.
But I don't think it's absolutely necessary.

Reply:
I made it positional, but had to change the argument order a bit. I also renamed it to filter, because it's a more fitting name.